### ISSUE TITLE
Not compiling on MacOSX

### ISSUE THREAD

**Issue by cphyc** [martinellimarco/indexed_zstd#2]
Your package is a dependency of https://github.com/mxmlnkn/ratarmount, which may become a dependency of https://github.com/yt-project/yt (if https://github.com/yt-project/yt/pull/3443 is merged). In the process of testing yt, we discovered that `indexed_zstd` does not compile on Mac-OSX (nor on Windows, but that is to be expected, isn't it?).

Unfortunately, I don't have access to a MacOS machine so I cannot debug it myself :(

# Steps to reproduce
```bash
# On MacOSX
pip install indexed_zstd
```
Actual error at build stage
```
[...]/indexed_zstd/libzstd-seek/zstd-seek.o -std=c++11 -O3 -DNDEBUG
    error: invalid argument '-std=c++11' not allowed with 'C'
    error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by martinellimarco** [comment]
Thank you for reporting this.

I never tested this on MacOS X before.
I did now and I confirm it's not working, but I get different error message.

I will work to fix this within the next week, in the meantime for completeness I'm adding the output I got.

Tested on Catalina 10.15.7, I will test on MacOS 11 later.

```
$ pip install indexed_zstd
DEPRECATION: Python 3.4 support has been deprecated. pip 19.1 will be the last one supporting it. Please upgrade your Python as Python 3.4 won't be maintained after March 2019 (cf PEP 429).
Collecting indexed_zstd
  Downloading https://files.pythonhosted.org/packages/f0/8f/97b8e250af5c2a41b0f0bc81ccc7399964f274026e5f93579f03a25dba8a/indexed_zstd-1.2.2.tar.gz (60kB)
     |████████████████████████████████| 450kB 22.0MB/s 
Building wheels for collected packages: indexed-zstd
  WARNING: Building wheel for indexed-zstd failed: [Errno 13] Permission denied: '/Users/imac/Library/Caches/pip/wheels/dd'
Failed to build indexed-zstd
Installing collected packages: indexed-zstd
  Running setup.py install for indexed-zstd ... error
    ERROR: Complete output from command /Library/Frameworks/Python.framework/Versions/3.4/bin/python3 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/t5/dz9b_c0s0cl6nd_tk8c2zrpw0000gn/T/pip-install-omxydj81/indexed-zstd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/t5/dz9b_c0s0cl6nd_tk8c2zrpw0000gn/T/pip-record-0ufzjkkc/install-record.txt --single-version-externally-managed --compile:
    ERROR: /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py:260: UserWarning: Unknown distribution option: 'long_description_content_type'
      warnings.warn(msg)
    running install
    running build
    running build_py
    file indexed_zstd.py (for module indexed_zstd) not found
    file indexed_zstd.py (for module indexed_zstd) not found
    running build_ext
    building 'indexed_zstd' extension
    creating build
    creating build/temp.macosx-10.6-intel-3.4
    creating build/temp.macosx-10.6-intel-3.4/indexed_zstd
    creating build/temp.macosx-10.6-intel-3.4/indexed_zstd/libzstd-seek
    /usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch i386 -arch x86_64 -g -I. -I/Library/Frameworks/Python.framework/Versions/3.4/include/python3.4m -c indexed_zstd/indexed_zstd.cpp -o build/temp.macosx-10.6-intel-3.4/indexed_zstd/indexed_zstd.o -std=c++11 -O3 -DNDEBUG
    clang: warning: include path for libstdc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    clang: warning: include path for libstdc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    indexed_zstd/indexed_zstd.cpp:655:10: fatal error: 'ios' file not found
    #include "ios"
             ^~~~~
    1 error generated.
    error: command '/usr/bin/clang' failed with exit status 1
    ----------------------------------------
ERROR: Command "/Library/Frameworks/Python.framework/Versions/3.4/bin/python3 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/t5/dz9b_c0s0cl6nd_tk8c2zrpw0000gn/T/pip-install-omxydj81/indexed-zstd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/t5/dz9b_c0s0cl6nd_tk8c2zrpw0000gn/T/pip-record-0ufzjkkc/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /private/var/folders/t5/dz9b_c0s0cl6nd_tk8c2zrpw0000gn/T/pip-install-omxydj81/indexed-zstd/
```

**Comment by martinellimarco** [comment]
I can also test this on Windows if it's needed but for me it's a low priority at the moment.

**Comment by martinellimarco** [comment]
Fixed in version 1.3.1, tested on Catalina 10.11.7 and Big Sur 11.2.3

```
$ pip3 install indexed_zstd --user
Collecting indexed_zstd
  Using cached https://files.pythonhosted.org/packages/d2/43/bd769e9207faa984362b39a2d1b7148f96c2e99a4473c6965c8618f86f4b/indexed_zstd-1.3.1.tar.gz
Building wheels for collected packages: indexed-zstd
  Building wheel for indexed-zstd (setup.py) ... done
  Created wheel for indexed-zstd: filename=indexed_zstd-1.3.1-cp38-cp38-macosx_10_14_6_x86_64.whl size=77046 sha256=5e84cd3d556b8a2fa0e61d4282ec2ed769e4096ad6ba49d8dbd5ac390125a8a3
  Stored in directory: /Users/imac/Library/Caches/pip/wheels/b2/79/d2/603d15751b915fed81135fd6cd2b3a496ecf7d7c79ce21e981
Successfully built indexed-zstd
Installing collected packages: indexed-zstd
Successfully installed indexed-zstd-1.3.1
```
